experimental/devices/inky/inky.go

```diff
@@ -188,84 +211,58 @@ func (d *Dev) Draw(dstRect image.Rectangle, src image.Image, srcPtrs image.Point
 	return d.update(borderColor[d.border], bufA, bufB)
 }
 
+// DrawAll redraws the whole display.
+func (d *Dev) DrawAll(src image.Image) error {
+	return d.Draw(d.Bounds(), src, image.ZP)
+}
+
 func (d *Dev) update(border byte, black []byte, red []byte) (err error) {
 	if err := d.reset(); err != nil {
 		return err
 	}
 
-	if err := d.sendCommand(0x74, []byte{0x54}); err != nil { // Set Analog Block Control.
-		return err
-	}
-	if err := d.sendCommand(0x7e, []byte{0x3b}); err != nil { // Set Digital Block Control.
-		return err
-	}
-
 	r := make([]byte, 3)
-	binary.LittleEndian.PutUint16(r, rows)
-	if err := d.sendCommand(0x01, r); err != nil { // Gate setting
-		return err
-	}
+	binary.LittleEndian.PutUint16(r, uint16(d.Bounds().Size().Y))
+	h := make([]byte, 4)
+	binary.LittleEndian.PutUint16(h[2:], uint16(d.Bounds().Size().Y))
 
-	init := []struct {
+	type cmdData struct {
 		cmd  byte
 		data []byte
-	}{
-		{0x03, []byte{0x10, 0x01}}, // Gate Driving Voltage.
-		{0x3a, []byte{0x07}},       // Dummy line period
-		{0x3b, []byte{0x04}},       // Gate line width
-		{0x11, []byte{0x03}},       // Data entry mode setting 0x03 = X/Y increment
-		{0x04, nil},                // Power on
-		{0x2c, []byte{0x3c}},       // VCOM Register, 0x3c = -1.5v?
+	}
+	cmds := []cmdData{
+		{0x01, r},                        // Gate setting
+		{0x74, []byte{0x54}},             // Set Analog Block Control.
+		{0x7e, []byte{0x3b}},             // Set Digital Block Control.
+		{0x03, []byte{0x17}},             // Gate Driving Voltage.
+		{0x04, []byte{0x41, 0xac, 0x32}}, // Gate Driving Voltage.
+		{0x3a, []byte{0x07}},             // Dummy line period
+		{0x3b, []byte{0x04}},             // Gate line width
+		{0x11, []byte{0x03}},             // Data entry mode setting 0x03 = X/Y increment
+		{0x2c, []byte{0x3c}},             // VCOM Register, 0x3c = -1.5v?
 		{0x3c, []byte{0x00}},
 		{0x3c, []byte{byte(border)}}, // Border colour
 	}
-
-	for _, c := range init {
-		if err := d.sendCommand(c.cmd, c.data); err != nil {
-			return err
-		}
+	if d.color == Yellow {
+		cmds = append(cmds, cmdData{0x04, []byte{0x07, 0xac, 0x32}}) // Set voltage of VSH and VSL
 	}
+	cmds = append(cmds, cmdData{0x32, modelLUT[d.color]}) // Set LUTs
```

Comment:
Does this have to be done after the VSH/VSL or can it be included in the block lines 233-244? Same for the rest.

Reply:
Yep it seems to still work with all these moved above the VSH/VSL command.  I think it's just the 0x22 (image update) that needs to go at the end .